dhcp-controller/api/v1alpha1/dhcpserver_types.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DHCPServerSpec defines the desired state of DHCPServer
+type DHCPServerSpec struct {
+	// Details of Servers
+	// +kubebuilder:validation:Required
+	Servers []Server `json:"servers,omitempty"`
+}
+
+type Server struct {
+	// refers to net-attach-def to be served
+	// +kubebuilder:validation:Required
+	NetworkName string `json:"networkName,omitempty"`
+	// refers to IP address to bind interface to
+	// +kubebuilder:validation:Required
+	InterfaceIp string `json:"interfaceIp,omitempty"`
+	// refers to CIDR of server
+	// +kubebuilder:validation:Required
+	ServerCIDR CIDR `json:"cidr,omitempty"`
+	// refers to leasetime of IP
+	// +kubebuilder:validation:Required
+	LeaseTime string `json:"leaseTime,omitempty"`
+	// refers to vlan
+	VlanID string `json:"vlanid,omitempty"`
+}
+
+// CIDR defines CIDR of each network
+type CIDR struct {
+	// refers to start IP of range
+	// +kubebuilder:validation:Required
+	CIDRIP string `json:"cidrIp,omitempty"`
+	// refers to start IP of range
+	RangeStartIp string `json:"rangeStartIp,omitempty"`
+	// refers to end IP of range
+	RangeEndIp string `json:"rangeEndIp,omitempty"`
```

Comment:
One more nitpick request: Can we make the json tags the same as whereabouts? That way the config is easy to port and remember

https://github.com/k8snetworkplumbingwg/whereabouts#core-parameters

They are range (for the cidr), range_start and range_end, and just "gateway". For example

```      
"ipam": {
        "type": "whereabouts",
        "range": "10.128.165.0/24",
        "range_start": "10.128.165.32",
        "range_end": "10.128.165.34",
        "gateway":10.128.165.1",
```

Reply:
Done